### PR TITLE
[Merged by Bors] - chore(Combinatorics/SimpleGraph): rename and add `gcongr` tag to distance lemmas

### PR DESCRIPTION
**Pull request by Rida-Hamadani**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by Rida-Hamadani** [comment]
@YaelDillies is it a good idea to tag the `dist` lemma with `gcongr` too?

File: Mathlib/Combinatorics/SimpleGraph/Metric.lean [hunk 1]
```diff
@@ -267,7 +268,7 @@ lemma dist_top [DecidableEq V] : (⊤ : SimpleGraph V).dist u v = (if u = v then
   by_cases h : u = v <;> simp [h]
 
 /-- Supergraphs have smaller or equal distances to their subgraphs. -/
-theorem dist_le_subgraph_dist {G' : SimpleGraph V} (h : G ≤ G') (hr : G.Reachable u v) :
+protected theorem Reachable.dist_anti {G' : SimpleGraph V} (h : G ≤ G') (hr : G.Reachable u v) :
     G'.dist u v ≤ G.dist u v := by
```
Comment (YaelDillies) [inline Mathlib/Combinatorics/SimpleGraph/Metric.lean:272]: Does `gcongr` accept this?
```suggestion
@[gcongr]
protected theorem Reachable.dist_anti {G' : SimpleGraph V} (h : G ≤ G') (hr : G.Reachable u v) :
    G'.dist u v ≤ G.dist u v := by
```

Comment (Rida-Hamadani) [reply Mathlib/Combinatorics/SimpleGraph/Metric.lean:272]: Yes!

**Comment by YaelDillies** [comment]
Let's go!

maintainer merge

File: Mathlib/Combinatorics/SimpleGraph/Metric.lean (hunk 1 above)
Comment (b-mehta) [reply Mathlib/Combinatorics/SimpleGraph/Metric.lean:273]: I'm surprised! Is it able to actually use this?

**Review by b-mehta — COMMENTED** [review]
Nice stuff

bors merge

**Comment by Ruben-VandeVelde** [comment]
Deprecated aliases seem to be missing?

**Comment by Rida-Hamadani** [comment]
@Ruben-VandeVelde Sorry, these lemmas were added recently, and so far only used in an open PR by me which I will edit. Should we add deprecated aliases or is it okay if we leave it this way?

**Pull request closed without merging**